zerver/webhooks/slack_incoming/view.py

```diff
@@ -25,9 +28,33 @@
 from zerver.lib.webhooks.common import OptionalUserSpecifiedTopicStr, check_send_webhook_message
 from zerver.models import UserProfile
 
+ParamT = ParamSpec("ParamT")
+
+
+def slack_error_handler(view_func: Callable[..., HttpResponse]) -> Callable[..., HttpResponse]:
+    """
+    A decorator that catches JsonableError exceptions and returns a
+    Slack-compatible error response in the format:
+    {ok: false, error: "error message"}.
+    """
+
+    @wraps(view_func)
+    def wrapped_view(
+        request: HttpRequest, *args: ParamT.args, **kwargs: ParamT.kwargs
+    ) -> HttpResponse:
+        try:
+            return view_func(request, *args, **kwargs)
+        except JsonableError as error:
+            return JsonResponse(
+                {"ok": False, "error": error.code.name}, status=error.http_status_code
```

Comment:
I think we want the human-readable error string, not the error `code` name, here?

Reply:
I'd suggested using the machine-readable code after reading through the Slack documentation that was linked in the issue description (https://api.slack.com/web#responses) since they note that their error messages are "a short machine-readable error code". But it's probably fine to use our error messages for this case.